File: DesktopSample/API/PageAndRDL/CreateReport/VB.NET/CreateReport/ReportsForm.vb
Imports System.IO
Imports System.Text
Imports GrapeCity.ActiveReports

Public Class ReportsForm

    Public Sub New()
        '必要なデザイナ変数です。
        Encoding.RegisterProvider(CodePagesEncodingProvider.Instance)
        InitializeComponent()
    End Sub

    Private Sub ReportsForm_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        ' ページレポートオブジェクトにレイアウトを読み込みます。
        Dim report As PageReport = LayoutBuilder.BuildReportLayout()
        ' ページレポートオブジェクトにデータソースを追加します。
        report = LayoutBuilder.AddDataSetDataSource(report)
        ' ストリームにページレポートオブジェクトを読み込みます。
        Dim reportStream As MemoryStream = LayoutBuilder.LoadReportToStream(report)

        arvMain.LoadDocument(reportStream, GrapeCity.Viewer.Common.DocumentFormat.Rdlx)
        report.Dispose()
        reportStream.Dispose()
    End Sub

End Class

File: DesktopSample/API/PageAndRDL/CreateReport/C#/CreateReport/ReportsForm.cs
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Windows.Forms;
using System.IO;
using System.Xml;
using GrapeCity.ActiveReports;

namespace ActiveReports.Samples.CreateReport
{
	public partial class ReportsForm : Form
	{
		public ReportsForm()
		{
			InitializeComponent();
		}

		private void ReportsForm_Load(object sender, EventArgs e)
		{
			// ページレポートオブジェクトにレイアウトを読み込みます。
			PageReport report = LayoutBuilder.BuildReportLayout();
			// ページレポートオブジェクトにデータソースを追加します。
			report = LayoutBuilder.AddDataSetDataSource(report);
			// ストリームにページレポートオブジェクトを読み込みます。
			MemoryStream reportStream = LayoutBuilder.LoadReportToStream(report);
			reportStream.Position = 0;
			// ストリームをViewerにロードします。
			viewer1.LoadDocument(reportStream, GrapeCity.Viewer.Common.DocumentFormat.Rdlx);

			report.Dispose();
			reportStream.Dispose();
		}
	}
}
